### PR TITLE
Ban dashes in miropt test file names

### PR DESCRIPTION
**Pull request by JakobDegen**
Dashes in miropt test file names are annoying - specifically, they cause the test files to not be sorted next to their outputs, which makes it very difficult to tell what goes with what. This PR changes all files to use underscores and adds a blessable tidy check that enforces this.

r? @jyn514 

This PR is obviously going to bitrot very quickly. Because of this, I'm going to wait at least 48 hours and until review comes back to allow people from @rust-lang/wg-mir-opt or elsewhere to lodge complaints if there are any. If not, I'll rebase the branch on Friday afternoon PST and ask that it be `r+`'d with `p=1` at that time.

### CODE REVIEW

File: src/tools/tidy/src/mir_opt_tests.rs [hunk 1]
```diff
@@ -35,3 +35,36 @@ pub fn check(path: &Path, bad: &mut bool) {
...
+
+fn check_dash_files(path: &Path, bless: bool, bad: &mut bool) {
+    for file in walkdir::WalkDir::new(&path.join("test/mir-opt"))
+        .into_iter()
+        .filter_map(Result::ok)
+        .filter(|e| e.file_type().is_file())
+    {
+        let path = file.path();
+        if path.extension() == Some("rs".as_ref()) {
+            if let Some(name) = path.file_name().map(|s| s.to_str()).flatten() {
```
Comment (jyn514) [inline src/tools/tidy/src/mir_opt_tests.rs:47]:
```suggestion
            if let Some(name) = path.file_name().and_then(|s| s.to_str()) {
```

**Review by jyn514 — APPROVED** [review]
The tidy code LGTM :) I'll let someone from WG-mir-opt say whether they want this change, but it makes sense to me.

**Comment by oli-obk** [comment]
This has annoyed me time and time again. Thank you for fixing it

**Comment by JakobDegen** [comment]
Surprised that survived even 9 hours without conflicts 😂

**Pull request merged**